ts/utils/hooks/bottomSheet.tsx

```diff
@@ -168,3 +175,62 @@ export const useIOBottomSheetModal = (
   );
   return { present, dismiss: dismissAll, bottomSheet };
 };
+
+/**
+ * Hook to generate a bottomSheet with a title, snapPoint and a component, that autosizes to the height of its content
+ * @param bottomSheetOptions
+ * @see {BottomSheetOptions}
+ * @param bottomPadding the bottom padding of the bottom sheet, defaul is 0
+ */
+export const useIOBottomSheetAutoresizableModal = (
+  {
+    component,
+    title,
+    snapPoint: originSnap,
+    footer,
+    onDismiss
+  }: BottomSheetOptions,
+  bottomPadding: number = 0
+) => {
+  const [snapPoint, setSnapPoint] = React.useState<number>(originSnap[0]);
+
+  const handleContentOnLayout = (event: LayoutChangeEvent) => {
+    const { height } = event.nativeEvent.layout;
+
+    setSnapPoint(bottomPadding + height);
```

Comment:
What do you think if you adds also the safe-area bottom spacing insets?

```typescript
import { useSafeAreaInsets } from 'react-native-safe-area-context';
```

Then add outside the `handleContentOnLayout` function this code snippet:
```typescript
const insets = useSafeAreaInsets();
```

And finally you can add it to the final value of the calculated snapPoint

```suggestion
    setSnapPoint(insets.bottom + bottomPadding + height);
```

Reply:
I think that if we handle the bottom inset from the safearea we can even avoid to handle a common bottom padding greater than zero, letting it handle to any specific usage of this common hook, what do you think?

Reply:
I think that anyway also with the bottom inset from the SafeArea, is nice to have also the default padding greater than zero.
Take the following example, I added the suggested code and into wallet home i replaced the bottom sheet hook with the autoresizable one with inside a component composed by a List with just "Metodo di pagamento" item. As you can see, the content starts too near to the bottom bar.

<p align="center">
<img src="https://github.com/pagopa/io-app/assets/34343582/70b5c433-4300-43c9-a760-6a5c0eabf096" width="250" />
</p>